llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -1161,6 +1161,14 @@ bool MemCpyOptPass::processMemCpyMemCpyDependence(MemCpyInst *M,
                      MSSA->getMemoryAccess(MDep), MSSA->getMemoryAccess(M)))
     return false;
 
+  // No need to create `memcpy(a <- a)`.
+  if (BAA.isMustAlias(M->getDest(), MDep->getRawSource())) {
```

Comment:
There is an odd mix of getDest and getRawSource here. I'd expect either both or neither to use getRaw...

Reply:
I changed to `getDest` and `getSource`.